Tests/Integration/DotNet/Azure/CosmosDb/CosmosDbPartitionedStorageTests.cs

```diff
@@ -0,0 +1,150 @@
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace IntegrationTests.Azure.CosmosDb
+{
+    [Trait("TestCategory", "Storage")]
+    [Trait("TestCategory", "CosmosDb Partitioned")]
+    public class CosmosDbPartitionedStorageTests : CosmosDbBaseTests, IClassFixture<CosmosDbPartitionedStorageFixture>
+    {
+        private readonly CosmosDbPartitionedStorageFixture _cosmosDbFixture;
+
+        public CosmosDbPartitionedStorageTests(CosmosDbPartitionedStorageFixture cosmosDbFixture)
+        {
+            _cosmosDbFixture = cosmosDbFixture;
+        }
+
+        [Fact]
+        public Task CreateItem()
+        {
+            return CreateItemTest(_cosmosDbFixture.Storage);
+        }
+
+        [Fact]
+        public Task UpdateItem()
+        {
+            return UpdateItemTest(_cosmosDbFixture.Storage);
+        }
+
+        [Fact]
+        public Task ReadUnknownItem()
+        {
+            return ReadUnknownItemTest(_cosmosDbFixture.Storage);
+        }
+
+        [Fact]
+        public Task DeleteItem()
+        {
+            return DeleteItemTest(_cosmosDbFixture.Storage);
+        }
+
+        [Fact]
+        public Task CreateItemWithSpecialCharacters()
+        {
+            return CreateItemWithSpecialCharactersTest(_cosmosDbFixture.Storage);
+        }
+
+        [Fact]
+        public async Task CreateItemWithNestingLimit()
+        {
+            async Task TestNestAsync(int depth)
+            {
+                // This creates nested data with both objects and arrays
+                static JToken CreateNestedData(int count, bool isArray = false)
+                    => count > 0
+                        ? (isArray
+                            ? new JArray { CreateNestedData(count - 1, false) } as JToken
+                            : new JObject { new JProperty("data", CreateNestedData(count - 1, true)) })
+                        : null;
+
+                var dict = new Dictionary<string, object>
+                {
+                    { "nestingLimit", CreateNestedData(depth) },
+                };
+
+                await _cosmosDbFixture.Storage.WriteAsync(dict);
+            }
+
+            // Should not throw
+            await TestNestAsync(127);
+
+            try
+            {
+                // Should either not throw or throw a special exception
+                await TestNestAsync(128);
+            }
+            catch (InvalidOperationException ex)
+            {
+                // If the nesting limit is changed on the Cosmos side
+                // then this assertion won't be reached, which is okay
+                Assert.Contains("recursion", ex.Message);
+            }
+        }
+
+        [Fact]
+        public async Task CreateItemWithDialogsNestingLimit()
+        {
+            async Task TestDialogNestAsync(int dialogDepth)
+            {
+                Dialog CreateNestedDialog(int depth) => new ComponentDialog(nameof(ComponentDialog))
```

Comment:
This function could be made static
```suggestion
                static Dialog CreateNestedDialog(int depth) => new ComponentDialog(nameof(ComponentDialog))
```